ecl/eclccserver/eclccserver.cpp

```diff
@@ -948,23 +1008,18 @@ class EclccCompileThread : implements IPooledThread, implements IErrorReporter,
         if (!workunit)
         {
             DBGLOG("Workunit %s no longer exists", wuid.get());
-            return;
+            return WUStateUnknown;
         }
 
         if (isContainerized())
         {
-            if (!useChildProcesses && !config->hasProp("@workunit"))
+            if (!useChildProcesses && !compileLocal)
             {
-                //If the timelimit for child processes is 0, or a workunit is explicitly defined as slow to compile
-                //then start the compile immediately using a K8s job.
-                if ((childProcessTimeLimit == 0) || workunit->getDebugValueBool("isComplexCompile", false))
-                {
-                    //NOTE: This call does not modify the workunit itself, so no need to commit afterwards
-                    workunit->setContainerizedProcessInfo("EclCCServer", getComponentConfigSP()->queryProp("@name"), k8s::queryMyPodName(), k8s::queryMyContainerName(), nullptr, nullptr);
-                    workunit.clear();
-                    compileViaK8sJob(true);
-                    return;
-                }
+                //NOTE: This call does not modify the workunit itself, so no need to commit afterwards
```

Comment:
I'm not sure what this comment means.. the line below is altering the workunit and the workunit.clear() will commit won't it?

Reply:
@ghalliday - still not sure re. this comment, is it misplaced? setContainerizedProcessInfo does alter the workunit..